cmd/kube-apiserver/app/server.go

```diff
@@ -283,7 +283,15 @@ func CreateKubeAPIServerConfig(
 		return nil, nil, nil, nil, err
 	}
 
-	if _, port, err := net.SplitHostPort(s.Etcd.StorageConfig.Transport.ServerList[0]); err == nil && port != "0" && len(port) != 0 {
+	removeSchemePrefix := func(url string) string {
```

Comment:
nit: function doesn't need to be nested and assigned to a var. Just make it a unexported top level function in this package? A test that shows this works as expected with and without a scheme would be great too.

Reply:
https://play.golang.org/p/T_HU4VohyW8    Here is the testing code and it will return an error "too many colons in address" with a scheme.